examples/pytorch/rgcn/entity_sample.py

```diff
@@ -1,167 +1,125 @@
-                        help="gpu")
-    parser.add_argument("--n-bases", type=int, default=-1,
-                        help="number of filter weight matrices, default: -1 [use all]")
-    parser.add_argument("--n-epochs", type=int, default=50,
-                        help="number of training epochs")
-    parser.add_argument("-d", "--dataset", type=str, required=True,
-                        choices=['aifb', 'mutag', 'bgs', 'am'],
-                        help="dataset to use")
-    parser.add_argument("--wd", type=float, default=5e-4,
-                        help="weight decay")
-    parser.add_argument("--fanout", type=str, default="4, 4",
-                        help="Fan-out of neighbor sampling")
-    parser.add_argument("--use-self-loop", default=False, action='store_true',
-                        help="include self feature as a special relation")
-    parser.add_argument("--batch-size", type=int, default=100,
-                        help="Mini-batch size")
+    parser.add_argument("--dataset", type=str, default="aifb",
+                        help="Dataset name ('aifb', 'mutag', 'bgs', 'am').")
     args = parser.parse_args()
-
-    print(args)
-    main(args)
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    print(f'Training with DGL built-in RGCN module with sampling.')
+
+    # load and preprocess dataset
+    if args.dataset == 'aifb':
+        data = AIFBDataset()
+    elif args.dataset == 'mutag':
+        data = MUTAGDataset()
+    elif args.dataset == 'bgs':
+        data = BGSDataset()
+    elif args.dataset == 'am':
+        data = AMDataset()
+    else:
+        raise ValueError('Unknown dataset: {}'.format(args.dataset))
+    g = data[0]
+    num_rels = len(g.canonical_etypes)
+    category = data.predict_category
+    labels = g.nodes[category].data.pop('labels').to(device)
+    train_mask = g.nodes[category].data.pop('train_mask')
+    test_mask = g.nodes[category].data.pop('test_mask')
+    # find target category and node id
+    category_id = g.ntypes.index(category)
+    g = dgl.to_homogeneous(g, edata=None)
+    node_ids = torch.arange(g.num_nodes())
+    target_idx = node_ids[g.ndata[dgl.NTYPE] == category_id]
+    # rename the fields as they can be changed by DataLoader
+    g.ndata['ntype'] = g.ndata.pop(dgl.NTYPE)
+    g.ndata['type_id'] = g.ndata.pop(dgl.NID)
+    # find the mapping (inv_target) from global node IDs to type-specific node IDs
+    inv_target = torch.empty((g.num_nodes(),), dtype=torch.int64).to(device)
+    inv_target[target_idx] = torch.arange(0, target_idx.shape[0], dtype=inv_target.dtype).to(device)
+    
+    # create RGCN model    
+    in_size = g.num_nodes() # featureless with one-hot encoding
+    out_size = data.num_classes
+    model = RGCN(in_size, 16, out_size, num_rels).to(device)
+    
+    # model training
+    print('Training...')
+    train(device, g, target_idx, labels, train_mask, model)
+    
+    # test the model
+    print('Testing...')
```

Comment:
For L114-115 and L118-119, same as `entity.py`

Reply:
Right. I will keep them consistent.